pkg/certs/cert-inspection/certgraphanalysis/deduplication.go

```diff
@@ -13,7 +15,14 @@ func deduplicateCertKeyPairs(in []*certgraphapi.CertKeyPair) []*certgraphapi.Cer
 			if currOut == nil {
 				panic("two")
 			}
-			if currOut.Name == currIn.Name {
+			// Two certificates with matching names
+			if len(currOut.Name) != 0 && len(currIn.Name) != 0 && currOut.Name == currIn.Name {
+				ret[j] = combineSecretLocations(ret[j], currIn.Spec.SecretLocations)
+				ret[j] = combineCertOnDiskLocations(ret[j], currIn.Spec.OnDiskLocations)
+				found = true
+			}
+			// Key modulus and cert modulus match - add key to cert
+			if len(currOut.Name) != 0 && len(currIn.Name) == 0 && currOut.Spec.CertMetadata.CertIdentifier.PubkeyModulus == currIn.Spec.CertMetadata.CertIdentifier.PubkeyModulus {
```

Comment:
are these two mutually exclusive cases (switch) or do one and do the other (as written now)

Reply:
Right, these are exclusive. Rewrote it to make use of `switch`